### PR TITLE
Add testcase for `django.autoreload.clean_files`

### PR DESCRIPTION
**Pull request by igniteflow**

### CODE REVIEW

File: tests/utils_tests/test_autoreload.py [hunk 1]
```diff
@@ -185,3 +187,42 @@ def test_check_errors_catches_all_exceptions(self):
             with self.assertRaises(Exception):
                 autoreload.check_errors(import_module)('test_exception')
         self.assertFileFound(filename)
+
+
+class CleanFilesTestCase(TestCase):
```
Comment (charettes) [inline tests/utils_tests/test_autoreload.py]: Use `SimpleTestCase` as this is not executing any queries.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by igniteflow** [comment]
I've rebased and pushed the changes to fix Python 3 compatibility, remove TestCase as suggested and fix the isort errors.  Apologies for deleting the history, the intention was to make it easier to review as one chunk.

File: tests/utils_tests/test_autoreload.py [hunk 2]
```diff
@@ -9,6 +10,18 @@
...
 from django.utils import autoreload
 from django.utils._os import npath
+from django.utils.translation import trans_real
+
+try:
+    import thread as _thread  # Python 2
+except ImportError:
+    import _thread  # Python 3
+
+try:
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: Use `from django.test import mock`

File: tests/utils_tests/test_autoreload.py [hunk 3]
```diff
@@ -9,6 +10,18 @@
 from django.test.utils import extend_sys_path
 from django.utils import autoreload
 from django.utils._os import npath
+from django.utils.translation import trans_real
+
+try:
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: Use `from django.utils.six.moves import _thread`

File: tests/utils_tests/test_autoreload.py [hunk 4]
```diff
@@ -185,3 +198,59 @@ def test_check_errors_catches_all_exceptions(self):
             with self.assertRaises(Exception):
                 autoreload.check_errors(import_module)('test_exception')
         self.assertFileFound(filename)
+
+
+class CleanFilesTestCase(SimpleTestCase):
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: `CleanFilesTests` (TestCase implies the class doesn't contain any tests and is meant to be used a test base).

File: tests/utils_tests/test_autoreload.py [hunk 5]
```diff
@@ -185,3 +198,59 @@ def test_check_errors_catches_all_exceptions(self):
...
+    TEST_MAP = {
+        # description: (input_filelist, expected_returned_filelist)
+        'falsies': ([None, False], []),
+        'pycs': (['myfile.pyc'], ['myfile.py']),
+        'pyos': (['myfile.pyo'], ['myfile.py']),
+        '$py.class': (['myclass$py.class'], ['myclass.py']),
+        'combined': (
+            [None, 'file1.pyo', 'file2.pyc', 'myclass$py.class'],
+            ['file1.py', 'file2.py', 'myclass.py']
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: include trailing comma

File: tests/utils_tests/test_autoreload.py [hunk 6]
```diff
@@ -185,3 +198,59 @@ def test_check_errors_catches_all_exceptions(self):
...
+    }
+
+    def _run_tests(self, mock_files_exist=True):
+        with mock.patch('django.utils.autoreload.os.path.exists', return_value=mock_files_exist):
+            for description, values in self.TEST_MAP.items():
+                filenames, expected_returned_filenames = values
+                self.assertEqual(
+                    autoreload.clean_files(filenames),
+                    expected_returned_filenames if mock_files_exist else [],
+                    msg='{} failed for [input filelist] {} [returned filelist]:'.format(
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: seems to be missing a placeholder at the end

File: tests/utils_tests/test_autoreload.py [hunk 7]
```diff
@@ -185,3 +198,59 @@ def test_check_errors_catches_all_exceptions(self):
...
+                        description, filenames, expected_returned_filenames
+                    ),
+                )
+
+    def test_files_exist(self):
+        """
+        If the file exists it any compiled files (pyc, pyo, $py.class)
+        should be transformed as their source files
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: Please use periods

File: tests/utils_tests/test_autoreload.py [hunk 8]
```diff
@@ -185,3 +198,59 @@ def test_check_errors_catches_all_exceptions(self):
...
+        """
+        self._run_tests()
+
+    def test_files_do_not_exist(self):
+        """
+        If the files do not exist they should not be in the returned filelist
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: exist,

File: tests/utils_tests/test_autoreload.py [hunk 9]
```diff
@@ -185,3 +198,59 @@ def test_check_errors_catches_all_exceptions(self):
...
+
+class ResetTranslationsTestCase(SimpleTestCase):
+
+    def test_resets_gettext(self):
+        gettext._translations = {'foo': 'bar'}
+        autoreload.reset_translations()
+        self.assertEqual(gettext._translations, {})
+
+    def test_resets_trans_real(self):
+        trans_real._translations = {'foo': 'bar'}
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: Is it safe to set these values without resetting them at the end of the test?

File: tests/utils_tests/test_autoreload.py [hunk 10]
```diff
@@ -185,3 +198,59 @@ def test_check_errors_catches_all_exceptions(self):
...
+        self._run_tests()
+
+    def test_files_do_not_exist(self):
+        """
+        If the files do not exist they should not be in the returned filelist
+        """
+        self._run_tests(mock_files_exist=False)
+
+
+class ResetTranslationsTestCase(SimpleTestCase):
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: TestCase -> Tests

File: tests/utils_tests/test_autoreload.py [hunk 11]
```diff
@@ -185,3 +198,59 @@ def test_check_errors_catches_all_exceptions(self):
...
+                    autoreload.clean_files(filenames),
+                    expected_returned_filenames if mock_files_exist else [],
+                    msg='{} failed for [input filelist] {} [returned filelist]:'.format(
+                        description, filenames, expected_returned_filenames
+                    ),
+                )
+
+    def test_files_exist(self):
+        """
+        If the file exists it any compiled files (pyc, pyo, $py.class)
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: exists, (add comma, chop "it")

File: tests/utils_tests/test_autoreload.py [hunk 12]
```diff
@@ -185,3 +189,69 @@ def test_check_errors_catches_all_exceptions(self):
...
+        trans_real._translations = {'foo': 'bar'}
+        trans_real._default = 1
+        trans_real._active = False
+        autoreload.reset_translations()
+        self.assertEqual(trans_real._translations, {})
+        self.assertIsNone(trans_real._default)
+        self.assertIsInstance(trans_real._active, _thread._local)
+
+    def setUp(self):
+        super(ResetTranslationsTests, self).setUp()
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: No need to call `super()` in setUp/tearDown.

File: tests/utils_tests/test_autoreload.py [hunk 13]
```diff
@@ -185,3 +189,69 @@ def test_check_errors_catches_all_exceptions(self):
...
+    def test_resets_trans_real(self):
+        trans_real._translations = {'foo': 'bar'}
+        trans_real._default = 1
+        trans_real._active = False
+        autoreload.reset_translations()
+        self.assertEqual(trans_real._translations, {})
+        self.assertIsNone(trans_real._default)
+        self.assertIsInstance(trans_real._active, _thread._local)
+
+    def setUp(self):
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: setUp/tearDown should go at the top of the class.

File: tests/utils_tests/test_autoreload.py [hunk 14]
```diff
@@ -1,14 +1,18 @@
...
 from django import conf
 from django.contrib import admin
-from django.test import SimpleTestCase, override_settings
+from django.test import SimpleTestCase, override_settings, mock
 from django.test.utils import extend_sys_path
 from django.utils import autoreload
 from django.utils._os import npath
+from django.utils.translation import trans_real
+from django.utils.six.moves import _thread
+
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: don't add this blank line.

File: tests/utils_tests/test_autoreload.py [hunk 15]
```diff
@@ -185,3 +189,69 @@ def test_check_errors_catches_all_exceptions(self):
...
+                    expected_returned_filenames if mock_files_exist else [],
+                    msg='{} failed for [input filelist] {} [returned filelist] {}'.format(
+                        description, filenames, expected_returned_filenames
+                    ),
+                )
+
+    def test_files_exist(self):
+        """
+        If the file exists, any compiled files (pyc, pyo, $py.class)
+        should be transformed as their source files
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: please use periods

File: tests/utils_tests/test_autoreload.py [hunk 16]
```diff
@@ -1,14 +1,18 @@
+import gettext
 import os
 import shutil
 import tempfile
 from importlib import import_module
 
 from django import conf
 from django.contrib import admin
-from django.test import SimpleTestCase, override_settings
+from django.test import SimpleTestCase, override_settings, mock
```
Comment (timgraham) [inline tests/utils_tests/test_autoreload.py]: sort mock before override_settings to fix isort build failure

**Pull request merged**